lib/galaxy_test/api/test_groups.py

```diff
@@ -107,7 +107,9 @@ def test_update(self):
             another_user_id = self.dataset_populator.user_id()
             another_role_id = self.dataset_populator.user_private_role_id()
         assert another_user_id is not None
-        update_response = self._put(f"groups/{group_id}", data={"user_ids": [another_user_id]}, admin=True, json=True)
+        update_response = self._put(
```

Comment:
That's a breaking API change, right ? This will likely need double checking in the UI and in bioblend.

Reply:
@mvdbeek  Yes, but that change does not affect the UI. I have removed the `delete_existing_assocs=True` argument from the 3 `set_entity_FOO_associations` methods of the `model/security.py` module (with FOO=group/user/role). The behavior of those method has not changed: existing associations are deleted; that's why no changes to the client code were necessary. There was only one place in the code base where one of those methods was called with `delete_existing_assocs=False` - and that was the api test, which is why I added the existing user, so that the test passes the full set of user ids that *should* be associated with the group after the update (which is what the UI does). That said, it is a change nevertheless. I'll make sure to add this to the release notes - would that be sufficient? @nsoranzo do we need to make any changes in bioblend? I've tested the UI, of course.

Reply:
It does look like a breaking change since `delete_existing_assocs=False` was used since this API endpoint was introduced in 2012.
On the other hand, the [BioBlend documentation](https://github.com/galaxyproject/bioblend/blob/main/bioblend/galaxy/groups/__init__.py#L112-L118) is actually consistent with what you are implementing here, and would fix a deficiency of the API already found in https://github.com/galaxyproject/galaxy/pull/18374#discussion_r1636578400
Currently BioBlend tests only check that adding a new user to an empty group works, so we would need to extend the tests to verify the impact of this change.

Reply:
This is also noted in a [TODO inline comment](https://github.com/galaxyproject/galaxy/blob/fbb99abe85d53e118c60dea9f916ed006a478106/lib/galaxy_test/api/test_groups.py#L138) in the api test.
I can add new API tests to test this for group, as well as API tests for testing basic create/delete/update for the user and role associations. That said, the unit db tests in this PR already [test this extensively](https://github.com/galaxyproject/galaxy/blob/fbb99abe85d53e118c60dea9f916ed006a478106/test/unit/data/model/db/test_security.py).